### DISCUSSION TITLE
Problem beim Einrichten via Backup

### DISCUSSION THREAD

**Discussion by wora55** [weltenwort/home-assistant-rct-power-integration#269] [Q&A]
Ich habe RCT_Power auf einem Server mittels HACS eingerichtet und alles hat wunderbar funktioniert. Eine Woche später habe ich mich entschieden auf einen Pi4 zu wechseln, habe dort homeassistant installiert und dann ein Full Backup des alten Servers genommen und damit die Einrichtung automatisiert. Bis auf RCT_Power hat das erfreulicherweise funktioniert.
Bei RCT_Power wurde das Custom-Directory angelegt und ein Teil der Files dort auch hineinkopiert. Dann hat es irgendwie aufgehört mit der Fehlermeldung "[homeassistant.setup] Setup failed for custom integration rct_power: Unable to import component: cannot import name 'FrameCRCMismatch' from 'rctclient.exceptions' (/usr/local/lib/python3.10/site-packages/rctclient/exceptions.py)"

Mit der kann ich wenig anfangen, da in der Installation mit Hassio/Hass OS dieses Directory gar nicht zu existieren scheint (mit 'Terminal' nicht findbar), auch nicht in der 'alten' Installation, in der RCT_Power funktioniert. 

Irgend welche Ideen, wie ich weitermachen sollte?

**Comment by weltenwort** [comment]
Hallo, @wora55 :wave: Ich habe auf #268 geantwortet. Kurz gesagt sieht es fast so aus als hätte der Mechanismus, mit dem HA die Abhängigkeiten von Integrationen installiert, nicht ganz richtig funktioniert. Die Logs geben uns hoffentlich Aufschluss.

**Comment by wora55** [comment]
Hallo Felix,
ich habe den gleichen Vorgang ein zweites Mail durchgeführt. Dieses Mal habe ich aber ein Partielles Backup als Grundlage verwendet, bei dem ich bewusst die Apps, die über HACS installiert wurden ausgeschlossen habe (in dem Fall nur rct_power).
Interessanterweise war der tar-files dieses Backups grösser als der des vollständigen Backups.


Nach Installation des partiellen Backups war dann überraschenderweise auch rct_power installiert und hat ohne weitere Maßnahmen funktioniert. 


Die Installation mit Homeassistant OS ist zwar sehr bequem - aber nur, wenn alles korrekt funktioniert. Wenn nicht, ist man selbst als Entwickler mit 40 Jahren Erfahrung (wenn auch nicht bzgl. Docker) ziemlich aufgeschmissen.


Danke für Deine Mühen
  Wolfgang

Felix Stürmer schrieb am Freitag, 13. Januar 2023 01:27:44 (+01:00):



Hallo, @wora55 👋 Ich habe auf #268 geantwortet. Kurz gesagt sieht es fast so aus als hätte der Mechanismus, mit dem HA die Abhängigkeiten von Integrationen installiert, nicht ganz richtig funktioniert. Die Logs geben uns hoffentlich Aufschluss.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.


-- 
Wolfgang Radke
Schubertstr. 6f
76344 Eggenstein

**Comment by weltenwort** [comment]
Ok, gut zu hören dass es schlussendlich funktioniert hat. Leider klingt das eher wie ein Problem mit der allgemeinen Backup Funktionalität jenseits des Einflussbereichs dieser Integration. Ich hatte auch schon ähnliche Probleme, weshalb ich immer mehrere Generationen des nächtlichen Backuparchives aufbewahre.
